dart/test/monster_test_my_game.example2_generated.dart

```diff
@@ -26,6 +27,37 @@ class Monster {
   String toString() {
     return 'Monster{}';
   }
+
+  MonsterT unPack() => MonsterT();
+
+  static int pack(fb.Builder fbBuilder, MonsterT object) {
+    if (object == null) return 0;
+    return object.pack(fbBuilder);
+  }
+}
+
+class MonsterT {
+
+  int pack(fb.Builder fbBuilder) => _pack(fbBuilder, HashMap<Object, int>());
+
+  /// Write the object to the [Builder] only once and reuse the same offset on
```

Comment:
this re-use of object offsets is not something that is supported by any of the other object API implemenations. I understand why you would want it to round-trip DAGs, but that is a niche feature that is expensive to implement, and undesirable for the vast majority of cases where data is tree-shaped. I would suggest to remove it. If we feel strongly we must have this functionality, I'd suggest to by default making these existing offsets `null`, and only do the sharing if the user explicitly requests it. But in order to offer the same functionality as the other languages, I'd prefer removing the feature entirely.

Reply:
I'd love to remove that as well, I've only implemented it because the previous `XyzObjectBuilder` implementation did the same with `getOrCreateOffset()` in the same place. The whole `pack()` method is based on the existing `ObjectBuilder` - my idea was to keep it as close as possible without re-thinking the decisions made in that implementation. 

Please let me know if you want to diverge in this case and I'll remove the code.

Reply:
OK, I'll remove it from the new code for now at least.